### PR TITLE
refactor: re-organize navbar & sidebar

### PR DESCRIPTION
**Pull request by SukkaW**
CLO-125

The PR changes:

- Re-organize the sidebar, now every category (`Docs`, `Changelog`, `API`) has its own sidebar
- Split changelog docs into one version per file
- Adding the doc structure for the Karpenter GCP Provider and the Karpenter Alibaba Cloud Provider
- Some minor changes:
  - Add the `packageManager` field to the `package.json`, enabling automatic package manager selection with [corepack](https://nodejs.org/api/corepack.html). This doesn't affect any contributor who didn't enable corepack.

**Screenshots**

img_v3_02g5_e81bfb0e-55e9-4c8b-9201-68e450e77a3g

### CODE REVIEW

File: sidebars.ts [hunk 1]
```diff
@@ -25,34 +25,84 @@ const sidebars: SidebarsConfig = {
...
+  // The collection of "Karpenter GCP - 0.1.0" documents.
+  ['karpeneterGcp_0.1.0_Sidebar']: [
+    /* We can add a new document to the "Karpenter GCP - 0.1.0" version. */
+    // 'karpenter/gcp/0.1.0/getting-started',
+
+    /** We can also add an accordion using "type" category */
     {
       type: 'category',
-      label: 'Tips',
+      label: 'Karpenter GCP Provider',
```
Comment (jwcesign) [inline sidebars.ts:70]: Let's only keep AlibabaCloud tab temporarily.

File: sidebars.ts [hunk 2]
```diff
@@ -25,34 +25,84 @@ const sidebars: SidebarsConfig = {
...
+    // }
+  ],
+
+  // The collection of "Karpenter Alibaba Cloud - 0.1.0" documents.
+  ['karpeneterAlicloud_0.1.0_Sidebar']: [
+    {
+      type: 'category',
+      label: 'Karpenter Alibaba Cloud Provider',
+      items: [
+        'karpenter/alicloud/0.1.0/getting-started',
```
Comment (jwcesign) [inline sidebars.ts:97]: I think now it's preview version, not 0.1.0

**Review by jwcesign — COMMENTED** [review]
Other lgtm

File: sidebars.ts (hunk 1 above)
Comment (jwcesign) [reply sidebars.ts:70]: GCP is a little bit far yet

File: sidebars.ts (hunk 2 above)
Comment (SukkaW) [reply sidebars.ts]: Change the folder name and the navbar dropdown item to `preview` and `Preview`. Done in [`2bf09f6` (#11)](https://github.com/cloudpilot-ai/docs/pull/11/commits/2bf09f6a0f28b36c1717ae9386409cee3b4f6a35)

File: sidebars.ts (hunk 1 above)
Comment (SukkaW) [reply sidebars.ts]: Commented GCP out in [`2bf09f6` (#11)](https://github.com/cloudpilot-ai/docs/pull/11/commits/2bf09f6a0f28b36c1717ae9386409cee3b4f6a35). We can bring it back anytime.

**Pull request merged**